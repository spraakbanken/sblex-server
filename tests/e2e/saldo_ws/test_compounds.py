import pytest
from fastapi import status
from httpx import AsyncClient
from syrupy.extensions.json import JSONSnapshotExtension


@pytest.fixture
def snapshot_json(snapshot):
    return snapshot.with_defaults(extension_class=JSONSnapshotExtension)


@pytest.mark.parametrize("segment", ["dväljas", "bo"])
@pytest.mark.asyncio
async def test_sms_json(segment: str, client: AsyncClient, snapshot_json) -> None:
    response = await client.get(f"/sms/json/{segment}")
    assert response.status_code == status.HTTP_200_OK
    assert response.json() == snapshot_json


@pytest.mark.parametrize("segment", ["dväljas", "bo"])
@pytest.mark.asyncio
async def test_sms_html(segment: str, client: AsyncClient, snapshot) -> None:
    response = await client.get(f"/sms/html/{segment}")
    assert response.status_code == status.HTTP_200_OK
    assert response.text == snapshot


@pytest.mark.parametrize("segment", ["dväljas", "bo"])
@pytest.mark.asyncio
async def test_sms_xml(segment: str, client: AsyncClient, snapshot) -> None:
    response = await client.get(f"/sms/xml/{segment}")
    assert response.status_code == status.HTTP_200_OK
    assert response.headers["content-type"] == "application/xml"
    assert response.text == snapshot
