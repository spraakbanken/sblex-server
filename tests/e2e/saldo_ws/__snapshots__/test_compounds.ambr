# serializer version: 1
# name: test_sms_html[bo]
  '''
  <!doctype html>
  <html lang="sv">
      <head>
          <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
          <link
              rel="shortcut icon"
              href="https://svn.spraakdata.gu.se/sb-arkiv/pub/images/saldo_icon.png"
          />
          <link
              rel="stylesheet"
              type="text/css"
              href="http://testserver/static/saldo.css"
          />
          <title>Sammansättningsanalys för &#39;bo&#39; [Avstängd]</title>
          
          <script
              src="https://unpkg.com/htmx.org@1.9.8"
              crossorigin="anonymous"
          ></script>
      </head>
  
      <body>
          <center>
              <p>
                  <a href="http://testserver/fl/html"
                      ><img
                          src="https://spraakbanken.gu.se/sites/spraakbanken.gu.se/files/img/saldo/saldo.gif"
                          align="top"
                          alt="SALDO: gå tillbaka till startsidan"
                  /></a>
              </p>
          </center>
          <div id="output_table">
  <h1>Denna tjänst är tillfälligt avstängd</h1>
  <p><b>Orsak:</b> Tjänsten är stängd på grund av prestandaproblem.</p>
  <p>Läs mer <a href="https://github.com/spraakbanken/sblex-server/issues/110">här</a>.</p>
  
  <h1>This service is temporarily disabled</h1>
  <p><b>Reason: </b>Disabled due to performance issues.</p>
  <p>Read more in this <a href="https://github.com/spraakbanken/sblex-server/issues/110">tracking issue</a>.
  </p>
  
  </div>
          <footer>
              <hr />
              <ul>
                  <li>
                      <a href="https://spraakbanken.gu.se/saldo"
                          >SALDOs projektsida</a
                      >
                  </li>
                  <li>
                      <a
                          href="https://svn.spraakdata.gu.se/repos/sblex/pub/saldo_instruktion.pdf"
                          >dokumentation (pdf)</a
                      >
                  </li>
              </ul>
          </footer>
      </body>
  </html>
  '''
# ---
# name: test_sms_html[dv\xe4ljas]
  '''
  <!doctype html>
  <html lang="sv">
      <head>
          <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
          <link
              rel="shortcut icon"
              href="https://svn.spraakdata.gu.se/sb-arkiv/pub/images/saldo_icon.png"
          />
          <link
              rel="stylesheet"
              type="text/css"
              href="http://testserver/static/saldo.css"
          />
          <title>Sammansättningsanalys för &#39;dväljas&#39; [Avstängd]</title>
          
          <script
              src="https://unpkg.com/htmx.org@1.9.8"
              crossorigin="anonymous"
          ></script>
      </head>
  
      <body>
          <center>
              <p>
                  <a href="http://testserver/fl/html"
                      ><img
                          src="https://spraakbanken.gu.se/sites/spraakbanken.gu.se/files/img/saldo/saldo.gif"
                          align="top"
                          alt="SALDO: gå tillbaka till startsidan"
                  /></a>
              </p>
          </center>
          <div id="output_table">
  <h1>Denna tjänst är tillfälligt avstängd</h1>
  <p><b>Orsak:</b> Tjänsten är stängd på grund av prestandaproblem.</p>
  <p>Läs mer <a href="https://github.com/spraakbanken/sblex-server/issues/110">här</a>.</p>
  
  <h1>This service is temporarily disabled</h1>
  <p><b>Reason: </b>Disabled due to performance issues.</p>
  <p>Read more in this <a href="https://github.com/spraakbanken/sblex-server/issues/110">tracking issue</a>.
  </p>
  
  </div>
          <footer>
              <hr />
              <ul>
                  <li>
                      <a href="https://spraakbanken.gu.se/saldo"
                          >SALDOs projektsida</a
                      >
                  </li>
                  <li>
                      <a
                          href="https://svn.spraakdata.gu.se/repos/sblex/pub/saldo_instruktion.pdf"
                          >dokumentation (pdf)</a
                      >
                  </li>
              </ul>
          </footer>
      </body>
  </html>
  '''
# ---
# name: test_sms_xml[bo]
  '''
  <?xml version="1.0" encoding="UTF-8"?>
  <message>This service is temporarily disabled <reason>Disabled due to performance issues.</reason>
  <issue>https://github.com/spraakbanken/sblex-server/issues/110</issue>
  </message>
  '''
# ---
# name: test_sms_xml[dv\xe4ljas]
  '''
  <?xml version="1.0" encoding="UTF-8"?>
  <message>This service is temporarily disabled <reason>Disabled due to performance issues.</reason>
  <issue>https://github.com/spraakbanken/sblex-server/issues/110</issue>
  </message>
  '''
# ---
